### PR TITLE
TST: Add pandas wheels to Travis build

### LINKED ISSUE #307: TST: add pandas 0.20.1 to matrix
would also drop the 0.16.2 & 0.18 (just leave 0.17, 0.19, 0.20) from the matrix.

### PR DESCRIPTION
**Pull request by gfyoung**
* Add pandas `0.20.1` and `master` to Travis.
* Drop `0.16.2` and Python 3.4 machines.

Closes #307.

### CODE REVIEW

**Comment by jreback** [comment]
needs a rebase

also why don't we xfail the failing tests for the moment, just to get a green build (and create an issue to revert this).

**Comment by gfyoung** [comment]
@jreback : Sure thing.  I think what I'll do is put together a PR to patch the tests that will also close #303.

**Comment by gfyoung** [comment]
@jreback : Also see my question about the availability of `0.20.1` on `conda` channels in the original issue <a href="https://github.com/pydata/pandas-datareader/issues/307#issuecomment-299947878">here</a>.  So we might be able to get all green for everything but the `0.20.1` and `master` machines for now.

**Comment by gfyoung** [comment]
@jreback : Rebase done, and ready to go.  Will put up test-patch PR after this.

**Comment by gfyoung** [comment]
@jreback : Rebase is done, but I want to fix up the `master` build because there is no reason to be installing `numpy` from source.

**Comment by jreback** [comment]
``conda install pandas -c conda-forge`` will get you the latest (defaults is usually a few days behind on a new release)

File: .travis.yml [hunk 1]
```diff
@@ -30,8 +54,18 @@ install:
...
   - conda info -a
-  - conda create -q -n test-environment python=$PYTHON pandas=$PANDAS coverage setuptools html5lib lxml pytest pytest-cov
+  - conda create -q -n test-environment python=$PYTHON coverage setuptools html5lib lxml pytest pytest-cov
   - source activate test-environment
+  # Check if we are installing the daily builds for pandas and numpy.
+  # If so, install pandas from conda (to install dependencies) before
+  # installing the correct versions.
+  - if [[ "$PANDAS" == "MASTER" ]]; then
+      conda install pandas;
+      PRE_WHEELS="https://7933911d6844c6c53a7d-47bd50c35cd79bd838daf386af554a83.ssl.cf2.rackcdn.com";
```
Comment (jreback) [inline .travis.yml]: ``pip install numpy pytz python-dateutil`` here, remove the ``conda install pandas`` and this should work.

**Comment by gfyoung** [comment]
@jreback : No longer an issue it seems.  Also, FYI, when I install from the daily builds, it also pulls in the daily `numpy` build too (so not necessary to install `numpy` first, but still useful to install dependencies).

**Comment by jreback** [comment]
> @jreback : No longer an issue it seems. Also, FYI, when I install from the daily builds, it also pulls in the daily numpy build too (so not necessary to install numpy first, but still useful to install dependencies).

see my comment.when using pip, you need to satisfy its deps first.

**Comment by gfyoung** [comment]
@jreback : Not necessarily, as `pip` takes care of that for you.  But will change nonetheless.

**Comment by gfyoung** [comment]
@jreback : I fixed the installation for the `master` machine.  Ready to merge now.

**Comment by jreback** [comment]
prob should say only supports >= 0.17.0 or something. can you add a whatsnew/0.4.0 (and add this).

**Comment by gfyoung** [comment]
@jreback : Sounds good will do.

**Pull request merged**